src/parse/ris_live.go

```diff
@@ -5,45 +5,266 @@ import (
 	"encoding/json"
 	"fmt"
 	"log"
+	"net/netip"
 	"os"
+	"time"
 
 	"github.com/gorilla/websocket"
 )
 
 // RIS Live websocket url
 const socketUrl = "ws://ris-live.ripe.net/v1/ws/"
 
-func main() {
+var done chan interface{}
+var interrupt chan os.Signal
 
-	// create websocket connection to ris live websocket
-	conn, _, err := websocket.DefaultDialer.Dial(socketUrl, nil)
-	if err != nil {
-		log.Fatal("Websocket connection error:", err)
-		os.Exit(1)
-	}
-	defer conn.Close()
+type RisMessageData struct {
+	Host   string `json:"host,omitempty"`
+	Prefix string `json:"prefix,omitempty"`
+}
+
+type RisMessage struct {
+	Type string          `json:"type"`
+	Data *RisMessageData `json:"data"`
+}
 
+func receiveHandler(msgChannel chan common.BGPMessage, conn *websocket.Conn) {
+	//keep reading in new message from connection
 	for {
-		//keep reading in new message from connection
+
+		//take in next msg from connection
 		_, message, err := conn.ReadMessage()
 		if err != nil {
 			log.Println("Websocket read error:", err)
 			return
 		}
 
-		bgpMsg, err := parseLiveMessage(message)
+		//parse message to data structure
+		bgpMsgs, err := parseLiveMessage(message)
 		if err != nil {
 			log.Println("Error parsing BGP message", err)
```

Comment:
Just a question - do we need to do anything special if there is an error reading a message? Or do we just wait till the next one comes in?

Reply:
messages are independent of each other, so if the parse fails it's okay to log and move on. This _should_ be very rare.